### PR TITLE
Pods 7551

### PR DESCRIPTION
**Pull request by grantrobson**

### CODE REVIEW

File: app/controllers/event1/member/ErrorDescriptionController.scala [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+  def onSubmit(waypoints: Waypoints): Action[AnyContent] = (identify andThen getData(eventType)).async {
+    implicit request =>
+      form.bindFromRequest().fold(
+        formWithErrors =>
+          Future.successful(BadRequest(view(formWithErrors, waypoints))),
+        value => {
+          val originalUserAnswers = request.userAnswers.fold(UserAnswers())(identity)
+          val updatedAnswers = value match {
+            case Some(v) => originalUserAnswers.setOrException(ErrorDescriptionPage, v)
```
Comment (cyeniceri) [inline app/controllers/event1/member/ErrorDescriptionController.scala:58]: We didn't use same flow so far in other controllers. Do we need to also change them as well? 
setOrException or removeOrException

File: test/journey/TestJourneySpec.scala [hunk 2]
```diff
@@ -91,6 +95,25 @@ class TestJourneySpec extends AnyFreeSpec with JourneyHelpers with ModelGenerato
...
+
+    startingFrom(EventSelectionPage)
+      .run(
+        submitAnswer(EventSelectionPage, Event1),
+        submitAnswer(HowAddUnauthPaymentPage, Manual),
+        submitAnswer(WhoReceivedUnauthPaymentPage, Member),
+        next,
+        submitAnswer(MembersDetailsPage, membersDetails.get),
+        submitAnswer(DoYouHoldSignedMandatePage, true),
+        submitAnswer(ValueOfUnauthorisedPaymentPage, false),
```
Comment (cyeniceri) [inline test/journey/TestJourneySpec.scala:109]: submitAnswer has pageMustBe inside, so we don't need to call it before.

File: conf/app.routes [hunk 3]
```diff
@@ -127,3 +127,9 @@ POST        /new-report/event-1-dummy-address                  controllers.addre
 
 GET         /new-report/event-1-employer-payment-nature        controllers.event1.employer.PaymentNatureController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints)
 POST        /new-report/event-1-employer-payment-nature        controllers.event1.employer.PaymentNatureController.onSubmit(waypoints: Waypoints ?= EmptyWaypoints)
+
+GET         /new-report/event-1-surcharge                      controllers.event1.SchemeUnAuthPaySurchargeMemberController.onPageLoad(waypoints: Waypoints ?= EmptyWaypoints)
+POST        /new-report/event-1-surcharge                      controllers.event1.SchemeUnAuthPaySurchargeMemberController.onSubmit(waypoints: Waypoints ?= EmptyWaypoints)
```
Comment (cyeniceri) [inline conf/app.routes]: these GET and POST are duplicated I guess from a merge issue.
`/new-report/event-1-surcharge`
I removed them.

**Review by cyeniceri — APPROVED** [review]

**Comment by cyeniceri** [comment]
@grantrobson 
I'm only wondering that why we have a different logic in controller for this flow. Other than that all is ok for me.

File: app/controllers/event1/member/ErrorDescriptionController.scala (hunk 1 above)
Comment (grantrobson) [reply app/controllers/event1/member/ErrorDescriptionController.scala:58]: We can do either this way or the other way I guess. Both work. I prefer this way as if user enters nothing then nothing will get added to user answers. Presumably we will get an empty string the other way. But really I don't think it matters which way we do it.

**Review by cyeniceri — APPROVED** [review]

**Pull request merged**